packages/smart-vaults/src/utils/LightSyncMultiSigner.sol

```diff
@@ -84,56 +57,53 @@ abstract contract LightSyncMultiSigner is MultiSigner {
     /// @dev reverts if validation fails.
     function _validateSignerSetUpdate(
         MultiSignerLib.MultiSignerStorage storage $_,
-        SignerSetUpdate memory signerUpdate_,
+        SignerSetUpdate memory signerSetUpdate_,
         uint256 nonce_
     )
         internal
         view
     {
         if (
             !MultiSignerSignatureLib.isValidSignature(
-                $_, _getSignerUpdateHash(signerUpdate_, nonce_), signerUpdate_.normalSignature
+                $_, _getSignerUpdateHash(signerSetUpdate_, nonce_), signerSetUpdate_.normalSignature
             )
-        ) revert SignerSetUpdateValidationFailed(signerUpdate_);
+        ) revert SignerSetUpdateValidationFailed(signerSetUpdate_);
     }
 
-    function _processSignerSetUpdate(SignerSetUpdate memory signerUpdate_) internal {
-        SignerUpdateParam[] memory updateParams = signerUpdate_.updateParams;
-        uint256 numUpdates = updateParams.length;
-        for (uint256 i; i < numUpdates; i++) {
-            _processSignerUpdateParam(updateParams[i]);
-        }
+    function _processSignerSetUpdate(SignerSetUpdate memory signerSetUpdate_) internal {
+        (bytes memory signer, uint8 index) = abi.decode(signerSetUpdate_.data, (bytes, uint8));
+        _addSigner(signer, index);
     }
 
-    function _processSignerSetUpdatesMemory(SignerSetUpdate[] memory signerUpdates_)
+    function _processSignerSetUpdatesMemory(SignerSetUpdate[] memory signerSetUpdates_)
         internal
         view
-        returns (bytes memory signers, uint8 threshold)
+        returns (bytes memory signerUpdates)
     {
         MultiSignerLib.MultiSignerStorage storage $ = _getMultiSignerStorage();
         uint256 nonce = $.nonce;
-        threshold = $.threshold;
-        uint256 numUpdates = signerUpdates_.length;
-        signers = new bytes(256 * MultiSignerSignatureLib.SIGNER_SIZE);
+
+        uint256 numUpdates = signerSetUpdates_.length;
+        signerUpdates = new bytes(66 * numUpdates);
+
+        uint256 memorySignerSet;
 
         for (uint256 i; i < numUpdates; i++) {
             _validateSignerSetUpdateMemory({
                 $_: $,
-                signerUpdate_: signerUpdates_[i],
-                signers_: signers,
-                threshold_: threshold,
+                signerSetUpdate_: signerSetUpdates_[i],
+                signerUpdates_: signerUpdates,
                 nonce_: nonce++
             });
-            (signers, threshold) = _processSignerSetUpdateMemory(signerUpdates_[i], signers, threshold);
+            memorySignerSet = _processSignerSetUpdateMemory(i, signerSetUpdates_[i], memorySignerSet, signerUpdates);
```

Comment:
i think `memorySignerSet` should probably be checked/set in `validate` vs `process`? (also debating whether the whole code feels a lot more readable w/o the concept; kind of odd the way it works. at a minimum, needs a more descriptive name i think)

Reply:
yeah I can see why it would be confusing, validate only checks if the signatures to make this update are valid and not if the update itself is valid